civihr_employee_portal/js/reports.js

```diff
@@ -203,10 +205,83 @@
 
         $(this).find('.pvtCheckContainer p').addClass('chr_custom-checkbox');
       }
+
+      // Altering Employee length of service data to show number of days into
+      // String format of Year Months and days of length of service.
+      // Finding the field called Employee length of service.
+      if($(this).find('h4').text().indexOf("Employee length of service (") >= 0 ) {
+         // Iterating through all the filter values.
+        $(this).find('p.chr_custom-checkbox').each(function(i){
+          // Skip the Select All checkbox from filters.
+          if (!$(this).hasClass("pvtFilterSelectAllWrap")){
+            // Fetching the Employee length of service from filter values.
+            var employee_length_service = $(this).find('span:first').text();
+            // Replacing with format string calling function as parameter.
+            $(this).find('span:first').text(that.formatLengthsOfService(employee_length_service));
+          }
+        });
+      }
     });
   };
 
   /**
+   * Processes results in people report view to format length of service for each
+   * contact in a human readable form.
+   *
+   * @param employee_length_service
+   *
+   * @returns string
+   *
+   */
+  HRReport.prototype.formatLengthsOfService = function (employee_length_service) {
+
+    // Getting current date.
+    var date_current = new Date();
+    // Calculating date difference from length of service for Employee using current date.
+    var utime_target = date_current.getTime() + employee_length_service * 86400 * 1000;
+    var date_target = new Date(utime_target);
+
+    // Getting difference of Dates in integer between current and target date.
+    var diff_year  = parseInt(date_target.getUTCFullYear() - date_current.getUTCFullYear());
+    var diff_month = parseInt(date_target.getUTCMonth() - date_current.getUTCMonth());
+    var diff_day   = parseInt(date_target.getUTCDate() - date_current.getUTCDate());
+
+    // Checking for days in month considering leap year.
+    var days_in_month = [31, (date_target.getUTCFullYear() % 4 ? 29 : 28), 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+    var date_string = "";
+
+    while(true) {
+      var date_noun;
+      date_string = "";
+
+      // Checking for singular or plural for year string.
+      diff_year > 1 ? date_noun = "s" : date_noun = "";
+      date_string += (diff_year > 0 ? diff_year + " year" + date_noun + " " : "");
+      if(diff_month < 0) {
+        diff_year -= 1;
+        diff_month += 12;
+        continue;
+      }
+
+      // Checking for singular or plural for month string.
+      diff_month > 1 ? date_noun = "s" : date_noun = "";
+      date_string += (diff_month > 0 ? diff_month + " month" + date_noun + " " : "");
+      if(diff_day < 0) {
+        diff_month -= 1;
+        diff_day += days_in_month[((11 + date_target.getUTCMonth()) % 12)];
```

Comment:
Why do we need this weird math here? AFAIK, getUTCMonth() returns a number from 0 to 11, which are exactly the number of items on the days_in_month array, so I believe this would be enough:
```javascript
diff_day += days_in_month[date_target.getUTCMonth()];
```